packages/ddex-frontend/src/components/Uploads.tsx

```diff
@@ -0,0 +1,131 @@
+  const [nextId, setNextId] = useState<number | undefined>(undefined)
+  const [prevId, setPrevId] = useState<number | undefined>(undefined)
+  const { data, error, isPending } = useUploads(statusFilter, nextId, prevId)
+
+  const handleNext = () => {
+    if (data && data.length) {
+      setNextId(data[data.length - 1].id)
+      setPrevId(undefined)
+    }
+  }
+
+  const handlePrev = () => {
+    if (data && data.length) {
+      setPrevId(data[0].id)
+      setNextId(undefined)
+    }
+  }
+
+  return (
+    <>
+      <h1>Uploads</h1>
+      <div>
+        <button
+          type="button"
+          className="inline-flex items-center gap-x-1.5 rounded-md bg-purple-500 px-2.5 py-1.5 text-sm font-semibold text-white shadow-sm hover:bg-purple-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-purple-500 mx-1"
+          onClick={() =>
+            setStatusFilter((curFilter) =>
+              curFilter === 'success' ? '' : 'success'
+            )
+          }
+        >
+          Success
+          {statusFilter === 'success' && <CheckCircleIcon aria-hidden="true" />}
+        </button>
+        <button
+          type="button"
+          className="inline-flex items-center gap-x-1.5 rounded-md bg-purple-500 px-2.5 py-1.5 text-sm font-semibold text-white shadow-sm hover:bg-purple-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-purple-500 mx-1"
+          onClick={() =>
+            setStatusFilter((curFilter) =>
+              curFilter === 'pending' ? '' : 'pending'
+            )
+          }
+        >
+          Pending
+          {statusFilter === 'pending' && <CheckCircleIcon aria-hidden="true" />}
+        </button>
+        <button
+          type="button"
+          className="inline-flex items-center gap-x-1.5 rounded-md bg-purple-500 px-2.5 py-1.5 text-sm font-semibold text-white shadow-sm hover:bg-purple-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-purple-500"
+          onClick={() =>
+            setStatusFilter((curFilter) =>
+              curFilter === 'error' ? '' : 'error'
+            )
+          }
+        >
+          Error
+          {statusFilter === 'error' && <CheckCircleIcon aria-hidden="true" />}
+        </button>
+      </div>
+      {isPending && <div>Loading...</div>}
+      {error && <div>Error: {error.message}</div>}
+      {data && (
+        <table>
+          <thead>
+            <tr>
+              <th>ID</th>
+              <th>From Zip File</th>
+              <th>Uploaded By</th>
+              <th>Uploaded At</th>
+              <th>Status</th>
+            </tr>
+          </thead>
+          <tbody>
+            {data.map((upload, index) => (
+              <tr key={index}>
+                <td>{upload.id}</td>
+                <td>{upload.from_zip_file}</td>
+                <td>{upload.uploaded_by}</td>
```

Comment:
can we actually populate this since we should know the oauth'd user at the time of upload

Reply:
i guess i could just store whatever id oauth shares for now. we'll want some real auth on the upload endpoint eventually, which i haven't worked out the details for